dashboard-api/internal/platform/server/handler/studies/list.go

```diff
@@ -31,31 +24,17 @@ func ListHandler(studyRepository studiesmanager.StudyRepository) gin.HandlerFunc
 			return
 		}
 
-		data := make([]studyReponse, 0, len(studies))
-		for _, study := range studies {
-			data = append(data, studyReponse{
-				Id:        study.ID(),
-				Name:      study.Name(),
-				Slug:      study.Slug(),
-				CreatedAt: study.CreatedAt(),
-			})
+		var nextCursor interface{} = nil
+		if len(studies) >= pagination.Number {
+			nextCursor = createNextCursor(pagination)
 		}
 
-		if len(studies) < pagination.Number {
-			ctx.JSON(http.StatusOK, gin.H{
-				"data": data,
-				"pagination": gin.H{
-					"nextCursor": nil,
-				},
-			})
-		} else {
-			ctx.JSON(http.StatusOK, gin.H{
-				"data": data,
-				"pagination": gin.H{
-					"nextCursor": createNextCursor(pagination),
-				},
-			})
-		}
+		ctx.JSON(http.StatusOK, gin.H{
+			"data": studies,
+			"pagination": gin.H{
+				"nextCursor": nextCursor,
```

Comment:
Huh, if this is an interface, how does gin serialize it? Is it using reflection? Curious. Don't we know the type of nextCursor? Isn't it an integer or something?

Reply:
It can be either a string or null. I explained in this comment -> https://github.com/vlab-research/vlab/pull/57#discussion_r746160090 why I ended up using an interface.

About gin, it relies on json.Marshal to serialize it, which in turns relies on reflection (```e.reflectValue(reflect.ValueOf(dataToSerialize), opts)```)

Reply:
```Go
type Pagination struct {
   NextCursor `json:"nextCursor,omitempty"`
}
```